src/persist-client/src/internal/state.rs

```diff
@@ -169,6 +169,62 @@ pub struct HandleDebugState {
     pub purpose: String,
 }
 
+/// Part of the updates in a Batch.
+///
+/// Either a pointer to ones stored in Blob or the updates themselves inlined.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+#[serde(tag = "type")]
+pub enum BatchPart<T> {
+    Hollow(HollowBatchPart<T>),
+}
+
+impl<T> BatchPart<T> {
+    pub fn encoded_size_bytes(&self) -> usize {
+        match self {
+            BatchPart::Hollow(x) => x.encoded_size_bytes,
+        }
+    }
+
+    // A value of None indicates an inline part.
+    pub fn key(&self) -> Option<&PartialBatchKey> {
```

Comment:
This feels like the trickiest method of the bunch... it foists the case analysis off on the caller and there are quite a few callers.

Some ideas:
1. A few places (`describe`, `InvalidUsage`) just want a user-interpretable identifier or description of the batch. A `fn printable_name(&self) -> &str` method that returned `"<inline>"` or something for an inline batch seems harmless and useful.
2. A few usages want to delete the key if it's present. `EncodedPart::fetch` unifies this logic for fetching; maybe we can cook up a similar helper for dropping?

(1 seems likely to me to be an improvement; I'm not sure about 2. Take or leave as you prefer!)

Reply:
Yeah, fair! Did both (1) and (2).

After doing that, there was only a handful of callers of `fn key` left and over half of them seemed more clear (to me) as a match on BatchPart, so I went ahead and removed this method entirely.

(All in the prep PR)